File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
    
public:
    string reverseVowels(string s) {
 

int start=0;
int end=s.length()-1;
int mid=start+(end-start)/2;
while(start<=end){
    
    while(((s[start]!='a' and s[start]!='A') and ( s[start]!='e' and s[start]!='E') and (s[start]!='i' and s[start]!='I') and ( s[start]!='o' and s[start]!='O') and (s[start]!='u' and s[start]!='U')) and start<=end){
        start++;
    }
    
     while(((s[end]!='a' and s[end]!='A') and (s[end]!='e' and s[end]!='E' )and (s[end]!='i' and s[end]!='I') and (s[end]!='o' and s[end]!='O') and (s[end]!='u' and s[end]!='U') ) and start<=end) {
        end--;
    }
    if(start<=end){
    swap(s[start],s[end]);
    start++;
    end--;}



}
        
return s;    }

};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
      StringBuilder sb=new StringBuilder();
      Stack<Character> stack=new Stack<>();    
        for(int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if(c=='a'||c=='e'|| c=='o'||c=='u'|| c=='i'||c=='A'||c=='E'|| c=='O'||c=='U'|| c=='I')
            {
               stack.push(c);//eeoe
            }  
        }
      
          for(int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
             if(c=='a'||c=='e'|| c=='o'||c=='u'|| c=='i'||c=='A'||c=='E'|| c=='O'||c=='U'|| c=='I')
            {
               sb.append(stack.pop());
            } 
            else
            sb.append(c) ;
        }
     
        return sb.toString();
    }
}